File: src/js/net/WebSocket/WebSocketClient/IWebSocketClientApplication.js
class IWebSocketClientApplication extends IApplication {
    constructor(){
        super();
        this.defineInterface("onOpen",this.onOpen);
        this.defineInterface("onMessage",this.onMessage);
        this.defineInterface("onError",this.onError);
        this.defineInterface("onClose",this.onClose);
        this.defineInterface("isWebSocketClientApplication",this.isWebSocketClientApplication);
        this.validateInterface();
    }
    isWebSocketClientApplication() { return true;}
}

File: src/Android/app/src/main/java/fr/billet/vincent/libnet/IWebSocketClientApplication.java
package fr.billet.vincent.libnet;

import okhttp3.Response;
import okhttp3.WebSocket;

public interface IWebSocketClientApplication {
    public void onOpen(String response);
    public void onMessage(String text);
    public void onClose(int code, String reason);
    public void onError(String message);
}
